brian2/core/base.py

```diff
@@ -100,9 +102,9 @@ def __init__(self, dt=None, clock=None, when='start', order=0, name='brianobject
         #: The scope key is used to determine which objects are collected by magic
         self._scope_key = self._scope_current_key
         
-        logger.debug("Created BrianObject with name {self.name}, "
-                     "clock={self._clock}, "
-                     "when={self.when}, order={self.order}".format(self=self))
+        # logger.debug("Created BrianObject with name {self.name}, "
```

Comment:
Ah, yes and no. The problem is that currently the `Clock` is a `BrianObject` itself. So during the creation of the `Clock`, the `BrianObject` initialiser wants to print this log message which mentions the clock of the object that has been constructed, but in the case of a `Clock` the `clock` attribute refers to itself, and the string representation of that will fail, because we are not finished constructing it yet...
It's a bit cumbersome to fix, but it's not difficult. That said, are we now sure that we want to keep the "advance the clock with a codeobject" approach? Because if not, then the `Clock` will no longer be a `BrianObject` and this problem disappears...